### DISCUSSION TITLE
Get REST API collection entry based on slug?

### DISCUSSION THREAD

**Discussion by benebene84** [statamic/cms#10740] [Q&A] [answered]
Hey everyone,

I tried to search for similar issues, but didn't really find anything.

I am trying to use Statamic as a headless cms and render the content in a Nextjs Frontend.
My idea was to set-up a dynamic route and use the slug to fetch the data via the Rest API from statamic collection. 

However as far as I can see, it requires the id of the entry. And if i create a new page, the id is a randomly created.
Is there a way to set the id to the slug or request entries based on the slug? Or do I have to get all entries and then filter?

Thanks
Benedikt

**Comment by duncanmcclean** [comment, marked as answer]
> Is there a way to set the id to the slug or request entries based on the slug? Or do I have to get all entries and then filter?

As you've discovered, the ["single entry"](https://statamic.dev/rest-api#entry) doesn't support getting entries with anything other than the entry ID. 

You need to get all the entries and filter by slug instead:

```
/api/collections/articles/entries?filter[slug:is]=awesome
```

```php
'collections' => [
    'articles' => [
        'allowed_filters' => ['title', 'slug'],
    ],
],
```

**Reply by benebene84** [reply]
Thanks a lot for the confirmation! This is what I ended up doing, just in case somebody searches for the same:

```tsx
export default async function Page({ params }: { params: { slug: string } }) {
  const res = await fetch(`${process.env.CMS_URL}/api/collections/pages/entries/?filter[slug:is]=${slug}`);
  const data = await res.json();
  return (
    // do sth. with the data
  )
}
```
